test/index.test.js

```diff
@@ -451,10 +439,11 @@ test('Should not add cors headers other than `vary` when origin is false', t =>
   }, (err, res) => {
     t.error(err)
     delete res.headers.date
-    t.strictEqual(res.statusCode, 200)
-    t.strictEqual(res.payload, '')
+    t.strictEqual(res.statusCode, 404)
+    t.strictEqual(res.payload, 'Not Found')
     t.deepEqual(res.headers, {
-      'content-length': '0',
+      'content-length': '9',
+      'content-type': 'text/plain',
```

Comment:
I don't understand this change. Can you please it a bit? It seems `origin: false` with `preflight: true` does not make sense anymore. I would not add the route.

Reply:
See our discussion above. Origin can be a function and dynamically determined for each request.

So you are correct that it does not make sense and I'm returning 404. This is clearly a bit hacky... :thinking: thinking about better solutions.

Reply:
@mcollina is this change ok now with the new implementation that does allow for `origin: false` with `preflight: true` (will return 404).